crates/ide-completion/src/tests/expression.rs

```diff
@@ -2036,3 +2036,30 @@ fn foo() {
         "#]],
     );
 }
+
+#[test]
+fn completion_associated_type() {
+    check(
+        r#"
+        pub trait FieldPaths {
+            type T;
+        }
+        pub struct Person;
+        pub struct PersonPaths;
+        impl FieldPaths for Person {
+            type T = PersonPaths;
+        }
+        impl PersonPaths {
+            pub const NAME: &'static str = "name";
+            pub fn get_default() -> String { String::new() }
```

Comment:
`String` isn't available in our tests (that's why you have `{unknown}`), you can use primitive types like `i32` or `&str`.